File: src/VBNetFx/FAC_Generic/SubsidiaryInfo.vb
Imports ActiveDev
Imports System.Collections.ObjectModel
Imports System.Data.SQLCLient

Public Class SubsidiaryInfo
    Inherits InfoItemBase

    Private myIDSubsidiary As Integer
    Private mySubsidiaryName As String
    Private myStreet As String
    Private myCity As String
    Private myZip As String
    Private myCountryCode As String
    Private myCountry As String
    Private myPrimaryPhone As String

    Private Shared sharedSName As String

    Shared Sub New()
        sharedSName = RegistryHelper.SubsidiarySubstitutionName
    End Sub

    Public Overrides ReadOnly Property DisplayName() As String
        Get
            Return mySubsidiaryName
        End Get
    End Property

    Public Overridable Property IDSubsidiary() As Integer
        Get
            Return Me.myIDSubsidiary
        End Get
        Set(ByVal value As Integer)
            Me.myIDSubsidiary = value
        End Set
    End Property

    <ADAutoReportColumn("Name Subsidiarität", -2, 0)> _
    Public Overridable Property SubsidiaryName() As String
        Get
            Return Me.mySubsidiaryName
        End Get
        Set(ByVal value As String)
            Me.mySubsidiaryName = value
        End Set
    End Property

    <ADAutoReportColumn("Straße", -2, 1)> _
    Public Overridable Property Street() As String
        Get
            Return myStreet
        End Get
        Set(ByVal value As String)
            myStreet = value
        End Set
    End Property

    <ADAutoReportColumn("Ort", -2, 3)> _
    Public Overridable Property City() As String
        Get
            Return myCity
        End Get
        Set(ByVal value As String)
            myCity = value
        End Set
    End Property

    <ADAutoReportColumn("PLZ", -2, 2)> _
    Public Overridable Property Zip() As String
        Get
            Return myZip
        End Get
        Set(ByVal value As String)
            myZip = value
        End Set
    End Property

    Public Overridable Property CountryCode() As String
        Get
            Return myCountryCode
        End Get
        Set(ByVal value As String)
            myCountryCode = value
        End Set
    End Property

    <ADAutoReportColumn("Land", -2, 4)> _
    Public Overridable Property Country() As String
        Get
            Return myCountry
        End Get
        Set(ByVal value As String)
            myCountry = value
        End Set
    End Property

    Public Overridable Property PrimaryPhone() As String
        Get
            Return myPrimaryPhone
        End Get
        Set(ByVal value As String)
            myPrimaryPhone = value
        End Set
    End Property

    Overrides Function ToString() As String
        Return Me.SubsidiaryName
    End Function

    Public Overrides ReadOnly Property DataID() As Integer
        Get
            Return IDSubsidiary
        End Get
    End Property
End Class

Public Class SubsidiaryInfoCollection
    Inherits KeyedCollection(Of Integer, SubsidiaryInfo)

    Sub New(ByVal ConnectionString As String)
        MyBase.New()
        Dim locConnection As New SqlConnection(FacessoGeneric.SQLConnectionString)
        Using locConnection
            locConnection.Open()
            Dim locCommand As New SqlCommand("SELECT * FROM [Subsidiaries] ORDER by [SubsidiaryName]", locConnection)
            Dim locDR As SqlDataReader = locCommand.ExecuteReader()
            Do While locDR.Read()
                Dim locSi As New SubsidiaryInfo()
                locSi.IDSubsidiary = locDR.GetInt32(locDR.GetOrdinal("IDSubsidiary"))
                locSi.SubsidiaryName = locDR.GetString(locDR.GetOrdinal("SubsidiaryName"))
                locSi.Street = locDR.GetString(locDR.GetOrdinal("Street"))
                locSi.Zip = locDR.GetString(locDR.GetOrdinal("Zip"))
                locSi.City = locDR.GetString(locDR.GetOrdinal("City"))
                locSi.CountryCode = locDR.GetString(locDR.GetOrdinal("CountryCode"))
                locSi.Country = locDR.GetString(locDR.GetOrdinal("Country"))
                locSi.PrimaryPhone = locDR.GetString(locDR.GetOrdinal("PrimaryPhone"))
                Me.Add(locSi)
            Loop
        End Using
    End Sub

    Protected Overrides Function GetKeyForItem(ByVal item As SubsidiaryInfo) As Integer
        Return item.IDSubsidiary
    End Function
End Class

File: src/MixNetFx/Facesso.Generic/SubsidiaryInfo.cs
using System.Collections.ObjectModel;
using System.Data.SqlClient;
using ActiveDev;

namespace Facesso
{
    public class SubsidiaryInfo : InfoItemBase
    {
        private int myIDSubsidiary;
        private string mySubsidiaryName;
        private string myStreet;
        private string myCity;
        private string myZip;
        private string myCountryCode;
        private string myCountry;
        private string myPrimaryPhone;

        private static string sharedSName;

        static SubsidiaryInfo()
        {
            sharedSName = RegistryHelper.SubsidiarySubstitutionName;
        }

        public override string DisplayName => mySubsidiaryName;

        public virtual int IDSubsidiary
        {
            get { return myIDSubsidiary; }
            set { myIDSubsidiary = value; }
        }

        [ActiveDev.ADAutoReportColumn("Name Subsidiarität", -2, 0)]
        public virtual string SubsidiaryName
        {
            get { return mySubsidiaryName; }
            set { mySubsidiaryName = value; }
        }

        [ActiveDev.ADAutoReportColumn("Straße", -2, 1)]
        public virtual string Street
        {
            get { return myStreet; }
            set { myStreet = value; }
        }

        [ActiveDev.ADAutoReportColumn("Ort", -2, 3)]
        public virtual string City
        {
            get { return myCity; }
            set { myCity = value; }
        }

        [ActiveDev.ADAutoReportColumn("PLZ", -2, 2)]
        public virtual string Zip
        {
            get { return myZip; }
            set { myZip = value; }
        }

        public virtual string CountryCode
        {
            get { return myCountryCode; }
            set { myCountryCode = value; }
        }

        [ActiveDev.ADAutoReportColumn("Land", -2, 4)]
        public virtual string Country
        {
            get { return myCountry; }
            set { myCountry = value; }
        }

        public virtual string PrimaryPhone
        {
            get { return myPrimaryPhone; }
            set { myPrimaryPhone = value; }
        }

        public override string ToString() => SubsidiaryName;

        public override int DataID => IDSubsidiary;
    }

    public class SubsidiaryInfoCollection : KeyedCollection<int, SubsidiaryInfo>
    {
        public SubsidiaryInfoCollection(string connectionString) : base()
        {
            var locConnection = new SqlConnection(FacessoGeneric.SQLConnectionString);
            using (locConnection)
            {
                locConnection.Open();
                var locCommand = new SqlCommand("SELECT * FROM [Subsidiaries] ORDER by [SubsidiaryName]", locConnection);
                SqlDataReader locDR = locCommand.ExecuteReader();
                while (locDR.Read())
                {
                    var locSi = new SubsidiaryInfo();
                    locSi.IDSubsidiary = locDR.GetInt32(locDR.GetOrdinal("IDSubsidiary"));
                    locSi.SubsidiaryName = locDR.GetString(locDR.GetOrdinal("SubsidiaryName"));
                    locSi.Street = locDR.GetString(locDR.GetOrdinal("Street"));
                    locSi.Zip = locDR.GetString(locDR.GetOrdinal("Zip"));
                    locSi.City = locDR.GetString(locDR.GetOrdinal("City"));
                    locSi.CountryCode = locDR.GetString(locDR.GetOrdinal("CountryCode"));
                    locSi.Country = locDR.GetString(locDR.GetOrdinal("Country"));
                    locSi.PrimaryPhone = locDR.GetString(locDR.GetOrdinal("PrimaryPhone"));
                    Add(locSi);
                }
            }
        }

        protected override int GetKeyForItem(SubsidiaryInfo item) => item.IDSubsidiary;
    }
}
